fastcrypto-zkp/src/bn254/zk_login_api.rs

```diff
@@ -169,3 +170,27 @@ fn verify_zk_login_proof_with_fixed_vk(
     Groth16::<Bn254>::verify_with_processed_vk(&GLOBAL_VERIFYING_KEY, public_inputs, proof)
         .map_err(|e| FastCryptoError::GeneralError(e.to_string()))
 }
+
+/// Verify that the given parameters (name, value, aud, iss and salt_hash) were used to generate the
+/// given address.
+pub fn verify_zk_login_id(
+    address: &[u8],
+    name: &str,
+    value: &str,
+    aud: &str,
+    iss: &str,
+    salt_hash: &str,
+) -> FastCryptoResult<bool> {
```

Comment:
why not FastCryptoResult<()>?
also below

Reply:
Why do you prefer that? I think it makes sense to distinguish between an error and a mismatch, but we've used `FastCryptoResult<()>` for the signatures so we can also use it here for consistency.

Reply:
yes, we use () for verifications, and in this case you can return InvalidProof